### PR TITLE
Apply smart quote selection to RateInput

### LINKED ISSUE #1656: Smart Quote for the Limit Order Widget
- Make a good quote/base token selection
- Make sure its respected when you invert the token selection
- Also, good to have if it is in Swap one too (but way less prior for this)

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #1656

1. When one of the tokens is stable-coin, then another token is `quote` (should be displayed in `Price per TOKEN`)
2. When there are no stable-coins in trade, then consider a token with the smallest amount as `quote`. (for 0.0005 WETH -> 3000 COW, WETH is quote)

https://user-images.githubusercontent.com/7122625/206747578-fff03901-0dec-4c1d-8ab6-8fecb6a147e2.mov

### CODE REVIEW

File: src/cow-react/modules/limitOrders/updaters/MarketPriceUpdater/index.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export function MarketPriceUpdater() {
   const { price, isLoading } = useGetInitialPrice()
   const prevPrice = usePrevious(price)
 
-  useEffect(() => {
+  useLayoutEffect(() => {
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/updaters/MarketPriceUpdater/index.tsx:19]: To avoid blinking of the price

**Review by anxolin — COMMENTED** [review]
It didn't wor for me

- I went to limit orders
- WETH was selected
- I selected DAI
- Price was expressed in WETH (not in DAI)


image

**Comment by shoom3301** [comment]
@anxolin thanks! Fixed

**Comment by elena-zh** [comment]
Hey @shoom3301 , I have found that COW-1INCH token pair has different price views in Gnosis chain: 
image

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

File: src/cow-react/modules/limitOrders/containers/RateInput/index.tsx [hunk 2]
```diff
@@ -78,6 +83,36 @@ export function RateInput() {
...
+    inputCurrency,
+    outputCurrency,
+    inputCurrencyAmount,
+    outputCurrencyAmount,
+    updateLimitRateState,
+  ])
+
+  // Reset isQuoteCurrencySet flag on currencies changes
+  useEffect(() => {
+    setIsQuoteCurrencySet(false)
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/containers/RateInput/index.tsx:119]: The order of `useEffect()` matters, so, there shouldn't be race conditions

**Pull request merged**